### ISSUE TITLE
Warning: Operand of null-aware operation '!' has type 'String' which excludes null.

### ISSUE THREAD

**Issue by aaronkelton** [fredeil/email-validator.dart#35]
# Before adding null check

When using the `TextFormField`'s `validator` parameter, passing the value to `EmailValidator.validate()` warns that I need to **Add a null check (!)**. Screenshot showing the warning with suggested fix.

<img width="703" alt="CleanShot 2021-11-16 at 08 03 39@2x" src="https://user-images.githubusercontent.com/6170077/141999635-b3599d10-1569-452b-a7db-3dd5ae69c64a.png">

# After adding null check (plus print statement)
However, if I add the null check **bang**, and try to print the result of `EmailValidator.validate(value!)`, I get a different warning:
```
Warning: Operand of null-aware operation '!' has type 'String' which excludes null.
```

<img width="716" alt="CleanShot 2021-11-16 at 08 04 57@2x" src="https://user-images.githubusercontent.com/6170077/141999832-1e6042e3-9267-4708-a090-f3f98c4034e3.png">

# Question
Is this warning something that should be fixed in this `email-validator` package? The Stack Overflow posts I've read for this warning from other packages indicated that a fix was necessary in the package's code. E.g. https://stackoverflow.com/questions/67389956/why-im-getting-this-error-warning-operand-of-null-aware-operation-has-type

**Comment by fredeil** [comment]
Are you using the latest version? This package has impemented null-safety for quite some time.

I’m not sure if others are experiencing the same issue or not. I will look into the stackoverflow post to see if this affects the code.. @aaronkelton

**Comment by aaronkelton** [comment]
Version 2.0.1

I wish I had a better understanding of Dart and null-safety to understand this error message. I can get the error to disappear by changing the print line to use double question marks instead of bang:
```dart
print(EmailValidator.validate(value ?? ''));
```

I don't know if:
1. I'm using `value!` incorrectly per Dart best practices, or
2. There is some change that should be made in this package, or
3. There is some needed change to the `TextFormField` widget.

Maybe banging a variable more than once is the issue. 🤷‍♂️ Thank you for considering my issue! ❤️
